### PR TITLE
Support bulk un/assignments to relay-groups

### PR DESCRIPTION
**Pull request by mpeck**
This adds support for passing a list of bundles to relay-groups instead of doing it one at a time.

resolves https://github.com/operable/cog/issues/536

### CODE REVIEW

File: lib/cog_api/fake/relay_groups.ex [hunk 1]
```diff
@@ -95,13 +95,18 @@ defmodule CogApi.Fake.RelayGroups do
...
-    bundle = Server.show_by_key(Bundle, :name, bundle_name)
+  def add_bundles(_, _, %Endpoint{token: nil}), do: Endpoint.invalid_endpoint
+  def add_bundles(%{name: name}, %{bundles: bundle_names}, %Endpoint{token: _}=endpoint) do
     relay_group = Server.show_by_key(RelayGroup, :name, name)
-    add_bundle(bundle, relay_group)
+    bundle_ids = Enum.map(bundle_names, &Server.show_by_key(Bundle, :name, &1))
+    |> Enum.map(&Map.fetch!(&1, :id))
+    add_bundles(relay_group.id, bundle_ids, endpoint)
+  end
+  def add_bundles(id, bundle_ids, %Endpoint{token: _}=endpoint) when is_list(bundle_ids) do
```
Comment (drapergeek) [inline lib/cog_api/fake/relay_groups.ex]: Is there a time that `bundle_ids` would't be a list?

File: lib/cog_api/fake/relay_groups.ex [hunk 2]
```diff
@@ -114,13 +119,18 @@ defmodule CogApi.Fake.RelayGroups do
...
+    bundle_ids = Enum.map(bundle_names, &Server.show_by_key(Bundle, :name, &1))
+    |> Enum.map(&Map.fetch!(&1, :id))
+    remove_bundles(relay_group.id, bundle_ids, endpoint)
+  end
+  def remove_bundles(id, bundle_ids, %Endpoint{token: _}=endpoint) when is_list(bundle_ids) do
+    Enum.map(bundle_ids, &remove_bundles(id, &1, endpoint))
+    |> List.last
   end
-  def remove_bundle(id, bundle_id, %Endpoint{token: _}) do
+  def remove_bundles(id, bundle_id, %Endpoint{token: _}) do
```
Comment (drapergeek) [inline lib/cog_api/fake/relay_groups.ex]: It looks like the `remove_bundles` without just a list of `bundle_ids` wouldn't work. Is that correct?

Comment (mpeck) [inline lib/cog_api/fake/relay_groups.ex]: No, you should still be able to pass a single `bundle_id` and it should work. When you pass a list this part of the function is called per entry.

File: lib/cog_api/fake/relay_groups.ex (hunk 1 above)
Comment (mpeck) [inline lib/cog_api/fake/relay_groups.ex]: Yes. This function recurses. When a single id is passed the next function head is invoked.

**Comment by mpeck** [comment]
So, if my calculations are correct :) my changes should affect you guys too much. They only thing that might cause some funkiness is that I changed the name of `relay_group_add_bundle` and `relay_group_remove_bundle` to `relay_group_add_bundles` and relay_group_remove_bundles` respectively. If that is a major deal we can add the old functions back and just have them call the new functions.

Of course I could have missed something, so any feedback is most appreciated.

**Comment by drapergeek** [comment]
I think you're exactly right. This is a great move.

I don't have much opinion on the `singular vs plural`. Either way means you're going to end up incorrect at some point! I think going plural is fine, it indicates what the function can do.

:shipit:

**Pull request closed without merging**